Android/app/src/main/java/io/github/project_travel_mate/mytrips/MyTripFriendNameAdapter.java

```diff
@@ -1,32 +1,56 @@
 package io.github.project_travel_mate.mytrips;
 
 import android.app.Activity;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.Looper;
 import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.util.Log;
+import android.view.ContextThemeWrapper;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.ArrayAdapter;
 import android.widget.ImageView;
 import android.widget.TextView;
+import android.widget.Toast;
 
 import com.squareup.picasso.Picasso;
 
+import java.io.IOException;
 import java.util.List;
 import java.util.Objects;
 
 import butterknife.BindView;
 import butterknife.ButterKnife;
 import io.github.project_travel_mate.R;
+import objects.Trip;
 import objects.User;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import static utils.Constants.API_LINK_V2;
+import static utils.Constants.EXTRA_MESSAGE_TRIP_OBJECT;
 
 class MyTripFriendNameAdapter extends ArrayAdapter<User> {
     private final Activity mContext;
     private final List<User> mUsers;
+    private Trip mTrip;
+    private String mFriendId;
+    private String mToken;
```

Comment:
Instead of getting the token as a parameter in the adapter, why don't you fetch it here only using sharedPreferences?